### ISSUE TITLE
Default metrics provided by `spring-boot-actuator` don't work

### ISSUE THREAD

**Issue by chillleader** [camunda-community-hub/spring-zeebe#296]
### Issue description

`MetricsRecorder` bean declaration in `spring-zeebe-starter` breaks the default metrics auto configurations provided by Spring Boot Actuator.
https://github.com/camunda-community-hub/spring-zeebe/blob/47e151387040920b30c8914297fe084bc37cb00c/client/spring-zeebe-starter/src/main/java/io/camunda/zeebe/spring/client/actuator/ZeebeActuatorConfiguration.java#L23

Apparently, the `MeterRegistry` bean is eagerly injected, which breaks the initialization of other metrics. As a result, for, example, the `jvm` metrics are missing in the actuator endpoint:
<img width="1054" alt="Screenshot 2022-12-15 at 13 46 25" src="https://user-images.githubusercontent.com/38818382/207862720-2fdfed7a-d28e-429f-92c3-2d0b4181ae95.png">

### Possible solution

* Inject `MeterRegistry` bean with `@Lazy`

Similar issues with example solutions:
* https://github.com/spring-projects/spring-boot/issues/26630
* https://stackoverflow.com/questions/52148543/spring-boot-2-actuator-doesnt-publish-jvm-metric/53119998#53119998

**Comment by berndruecker** [comment]
Awesome - can you provide a PR @chillleader?

**Comment by chillleader** [comment]
Yes, I can provide an immediate fix by adding `@Lazy` and maybe also take a concentrated look at this later while working on the new connector metrics.

**Comment by berndruecker** [comment]
great stuff - from my perspective we can close this for now - I made a remark in #275 to have another look at it then

**Comment by chillleader** [comment]
I encountered this again in `8.2.0-alpha1`.

**Comment by berndruecker** [comment]
I can have a look at it - I think I have ideas why. Thanks for raising it!

**Comment by berndruecker** [comment]
Ok - after trying out different things I have to confess I don't fully understand the problem ;-) 

The fix proposed in https://github.com/spring-projects/spring-boot/issues/26684 also works for us, so if I add the following configuration, I have all metrics back - even without `@Lazy`.

```java
  @Bean
  InitializingBean forceMeterRegistryPostProcessor(BeanPostProcessor meterRegistryPostProcessor, MeterRegistry registry) {
    return () -> meterRegistryPostProcessor.postProcessAfterInitialization(registry, "");
  }
```

There were further quite some discussions/problems with such things with various Spring Boot 2.x versions. My proposal is that we simply apply this workaround and do not try to understand the big picture fully.

**Comment by berndruecker** [comment]
![grafik](https://user-images.githubusercontent.com/1217596/224950243-e994cd01-c285-413b-91fa-aad918d1b0e3.png)

**Comment by berndruecker** [comment]
@chillleader Can you probably double check with latest main? Should be fixed with https://github.com/camunda-community-hub/spring-zeebe/commit/16035396e5a089d75e2067d2106d4ffedadd7660

**Comment by chillleader** [comment]
Works like a charm ⭐  Thank you Bernd.
I'm also fine with this workaround.

**Comment by 445990772** [comment]
i have the same problem,
spring-boot version : 2.7.8

when I debuged MeterRegistryPostProcessor#postProcessAfterInitialization,
debuger not jump into if(bean instance of MeterRegistry) ，then i found jvmMetrics does not bind to PrometheusMeterRegistry
image

then i try to solve the problem with this , but it doesn't works ,MeterRegistryPostProcessor can not be imported cause it's not public
what else can i do ?

Btw. for anyone else having this issue, this fixes it for now:

@Configuration
public class FixPrometheusMetrics {
    public FixPrometheusMetrics(MeterRegistryPostProcessor postProcessor, PrometheusMeterRegistry registry) {
        postProcessor.postProcessAfterInitialization(registry, "");
    }
}

**Comment by 445990772** [comment]
i have the same problem,
spring-boot version : 2.7.8

when I debuged MeterRegistryPostProcessor#postProcessAfterInitialization,
debuger not jump into if(bean instance of MeterRegistry) ，then i found jvmMetrics does not bind to PrometheusMeterRegistry
![image](https://github.com/camunda-community-hub/spring-zeebe/assets/9440839/c939ad38-f2a4-4e56-ab8b-d68bb836a941)

then i try to solve the problem with this , but it doesn't works ,MeterRegistryPostProcessor can not be imported cause it's not public
what else can i do ?



> Btw. for anyone else having this issue, this fixes it for now:
> 
> @Configuration
> public class FixPrometheusMetrics {
>     public FixPrometheusMetrics(MeterRegistryPostProcessor postProcessor, PrometheusMeterRegistry registry) {
>         postProcessor.postProcessAfterInitialization(registry, "");
>     }
> }
>
